File: 3 - Fibonacci Sequence/3 - Fibonacci Sequence.py
def fibonacci_sequence(start_sequence, length):
    start, end = start_sequence
    final_sequence = start_sequence[:]

    if length == 0: return []
    if length == 1: return [start]
    if length == 2: return final_sequence
    
    for i in range(2, length):
        temp = start + end
        start, end = end, temp
        final_sequence.append(temp)
    
    return final_sequence

File: 3 - Fibonacci Sequence/3 - Fibonacci Sequence.js
function fibonacciSequence(startSequence, length) {
  let [start, end] = startSequence;

  if (length === 0) return []
  if (length === 1) return [start]
  if (length === 2) return startSequence

  let finalSequence = [...startSequence]
  for (let i = 2; i < length; i++) {
    let temp = start + end;
    start = end;
    end = temp;
    finalSequence[i] = temp;
  }

  return finalSequence;
}

console.log(fibonacciSequence([0, 1], 20))